lib/SILOptimizer/Analysis/ArraySemantic.cpp

```diff
@@ -710,9 +710,20 @@ bool swift::ArraySemanticsCall::replaceByValue(SILValue V) {
 
   SILBuilderWithScope Builder(SemanticsCall);
   auto &ValLowering = Builder.getTypeLowering(V->getType());
+
+  // In OSSA, the InsertPt is after V's definition and not before SemanticsCall
+  // Because we are creating copy_value in ossa, and the source may have been
+  // taken previously. So our insert point for copy_value is immediately after
+  // V, where we can be sure it is live.
+  auto InsertPt = V->getFunction()->hasOwnership()
```

Comment:
Why not just this to always do this? Then we have one implementation? I haven't thought through it maybe there is a reason not to.

Reply:
We could, but the retain would be much earlier than when we need, and I was thinking it may effect arc optimizations.